### PR TITLE
Adding ha metrics for failover detection

### PR DESCRIPTION
**Pull request by galcorlo**
Resolve: https://github.com/bluecmd/fortigate_exporter/issues/177

Already added is_root_master just in case because checking the
dashboard widget it seems we only need is_manage_master

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,9 @@ Global:
    * `fortigate_cpu_usage_ratio`
    * `fortigate_memory_usage_ratio`
    * `fortigate_current_sessions`
+ * _System/ha-checksums_
+   * `fortigate_is_master`
+   * `fortigate_serial_no`
```
Comment (bluecmd) [inline README.md:43]:
```suggestion
 * _System/HAChecksums_
   * `fortigate_ha_member_has_role`
```

File: pkg/probe/ha-checksum.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+type HAChecksum struct {
+	Results []HAChecksumResults `json:"results"`
+}
+
+func probeSystemHAChecksum(c http.FortiHTTP, meta *TargetMetadata) ([]prometheus.Metric, bool) {
+	var (
+		IsMaster = prometheus.NewDesc(
+			"fortigate_ha_role",
```
Comment (bluecmd) [inline pkg/probe/ha-checksum.go:23]: I think the metric name `fortigate_ha_member_has_role` is more aligned with the other HA metrics, thoughts?

File: pkg/probe/ha-checksum.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			"Master/Slave information",
+			[]string{"name", "serial_no"}, nil,
```
Comment (bluecmd) [inline pkg/probe/ha-checksum.go:25]:
```suggestion
			[]string{"role", "serial"}, nil,
```
The other metrics call it `serial`. `name` is a bit ambiguous, I think `role` is better.

File: pkg/probe/ha-checksum.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+	var res HAChecksum
+	if err := c.Get("api/v2/monitor/system/ha-checksums", "scope=global", &res); err != nil {
+		log.Printf("Error: %v", err)
+		return nil, false
+	}
+
+	m := []prometheus.Metric{}
+	for _, response := range res.Results {
+		m = append(m, prometheus.MustNewConstMetric(IsMaster, prometheus.GaugeValue, float64(response.IsManageMaster), "is_manage_master", response.SerialNo))
```
Comment (bluecmd) [inline pkg/probe/ha-checksum.go:37]: What do you think about dropping the "is_" part?

File: pkg/probe/ha-checksum.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package probe
```
Comment (bluecmd) [inline pkg/probe/ha-checksum.go:1]: File should be named system_ha_checksum.go to match the API it uses.

**Review by bluecmd — COMMENTED** [review]
Thanks for your contribution!

File: pkg/probe/ha-checksum.go (hunk 2 above)
Comment (galcorlo) [reply pkg/probe/ha-checksum.go:23]: Totally agree

File: pkg/probe/ha-checksum.go (hunk 3 above)
Comment (galcorlo) [reply pkg/probe/ha-checksum.go:25]: it make sense

File: pkg/probe/ha-checksum.go (hunk 5 above)
Comment (galcorlo) [reply pkg/probe/ha-checksum.go:1]: Perfect!

File: pkg/probe/ha-checksum.go (hunk 4 above)
Comment (galcorlo) [reply pkg/probe/ha-checksum.go:37]: I agree too

**Comment by galcorlo** [comment]
I agree with all your modifications :)

**Comment by bluecmd** [comment]
> I agree with all your modifications :)

Nice! Ping me when you have made the modifications and I'll approve and merge

**Review by galcorlo — COMMENTED** [review]
Comply with current naming

**Comment by galcorlo** [comment]
Hello Christian, not sure if I accepted all your proposed modifications. I'm a bit overwhelmed with the github interface/workflow

Thanks!

**Comment by bluecmd** [comment]
> Hello Christian, not sure if I accepted all your proposed modifications. I'm a bit overwhelmed with the github interface/workflow
> 
> Thanks!

I think what you did was closing my suggestions without merging them - and that's totally fine, but then you have to manually apply them.

**Review by galcorlo — COMMENTED** [review]
Files changed

**Comment by tiagoasousa** [comment]
Hi, any ETA for this to be merged? thanks!

**Review by bluecmd — APPROVED** [review]

**Pull request merged**